File: 2107-find-unique-binary-string/find-unique-binary-string.js
/**
 * @param {string[]} nums
 * @return {string}
 */
var findDifferentBinaryString = function(nums) {
    
    const set = new Set(nums);
    let ans = ""

    function check(str) {

        if(ans.length) return;
        if(str.length === nums[0].length) {

            if(!set.has(str)) ans = str;
            return
        }



        return check(str + "0") + check(str + "1")
    }

    check("")

    return ans;
};

File: 2107-find-unique-binary-string/find-unique-binary-string.cpp
class Solution {

    string check(string str, string & ans , vector<string>& nums) {
            if(ans.length() != 0) return  ans;
            if(str.length() == nums[0].length()) {

                for(int i = 0; i < nums.size(); i++) {
                    if(nums[i] == str) {
                       
                        return "";
                    }
                }
                ans = str;
                return ans;
            }
            str.push_back('0');
            check(str, ans, nums);
            str.pop_back();
            str.push_back('1');
            check(str, ans, nums);
            return ans;
        }

public:
    string findDifferentBinaryString(vector<string>& nums) {
        
        string ans = "";
        return check("", ans, nums);
        
    }
};